openedx/core/djangoapps/user_api/models.py

```diff
@@ -26,27 +28,56 @@ class Meta:  # pylint: disable=missing-docstring
 
     @classmethod
     def set_preference(cls, user, preference_key, preference_value):
+        """Sets the user preference for a given key, creating it if it doesn't exist.
+
+        Arguments:
+            user (User): The user whose preference should be set.
+            preference_key (string): The key for the user preference.
+            preference_value (object): The value to be stored.
+
+        Raises:
+            ValidationError: the update was rejected because it was invalid
         """
-        Sets the user preference for a given key
-        """
-        user_pref, _ = cls.objects.get_or_create(user=user, key=preference_key)
-        user_pref.value = preference_value
-        user_pref.save()
+        user_preference, _ = cls.objects.get_or_create(user=user, key=preference_key)
+        user_preference.value = preference_value
+        user_preference.full_clean()
```

Comment:
I've been going back and forth on this. It seems better to ensure that we don't save invalid values to the database which is why I left it in. However, the only validation that it really does is checking the key length, and that will be thrown as a database error (I imagine). What's your opinion?